core/src/test/scala/shapeless/generic.scala

```diff
@@ -122,6 +122,10 @@ package GenericTestsAux {
   sealed trait OB extends Overlapping
   case class OBC(s: String) extends OB
   case class OAB(i: Int) extends OA with OB
+
+  case object SelfGen {
+    val gen = Generic[SelfGen.type]
```

Comment:
can you please make this `implicit`

Reply:
Sure, although it won't change anything.